### PR TITLE
feat: integrate new list section (DEV-3545)

### PR DESCRIPTION
**Pull request by Nora-Olivia-Ammann**

### CODE REVIEW

File: src/dsp_tools/commands/excel2json/lists_compliance_checks.py
Comment (Nora-Olivia-Ammann) [inline src/dsp_tools/commands/excel2json/lists_compliance_checks.py:1]: Due to circular imports, this file was merged with the new_lists.py

Comment (BalduinLandolt) [reply src/dsp_tools/commands/excel2json/lists_compliance_checks.py:1]: as a rule of thumb, in case of circular dependencies, I would generally advise against merging code (because this doesn't inherently address the issue of the circular dependency, but just widens the code unit to include the entire circle), but instead try to extract the shared logic into a third unit, upon which both other units may depend without forming a circular reference. It may be ok in this instance, hard to say spontaneously, but just as a general remark.

File: src/dsp_tools/commands/excel2json/project.py
Comment (BalduinLandolt) [inline src/dsp_tools/commands/excel2json/project.py:97]: is this double colon there intentionally?

**Review by BalduinLandolt — COMMENTED** [review]
I only had a brief look. From what I can see in code, it looks ok.

However it is not exactly what we agreed upon in today's meeting: We said we'd accept the shortcut of not attempting to follow all quality assurance we should normally do, in order to get this finished out there, so you can prioritise other work without having to keep the branch open and the feature half-finished; and we agreed that this is acceptable, if we mark this as an experimental feature and consider the old workflow to be still the stable and default one. Now, renaming the old workflow to "old-xyz" and changing the behaviour of an established workflow sends a very different message, makes this a breaking change (which it would not be otherwise) and therefor goes against the idea of trying to improve the perceived quality of our software.  
In the backend, we embraced the policy of not implementing any breaking changes without deprecating the old behaviour and giving users a grace period of some weeks.

I think I would like this PR a lot better, if it left `excel2lists` and `excel2json` unchanged, and exposed either `excel2lists-experimental` or `excel2lists-new` or `excel2lists --experimental` or something like this.  
And just to be clear: The naming in code, I'm absolutely fine with. I'm really only referring to the naming used in the parser, which defines the CLI that is exposed to the users.

File: src/dsp_tools/commands/excel2json/lists_compliance_checks.py
Comment (Nora-Olivia-Ammann) [reply src/dsp_tools/commands/excel2json/lists_compliance_checks.py:1]: Thanks, that is very valid.

**Comment by Nora-Olivia-Ammann** [comment]
> I only had a brief look. From what I can see in code, it looks ok.
> 
> However it is not exactly what we agreed upon in today's meeting: We said we'd accept the shortcut of not attempting to follow all quality assurance we should normally do, in order to get this finished out there, so you can prioritise other work without having to keep the branch open and the feature half-finished; and we agreed that this is acceptable, if we mark this as an experimental feature and consider the old workflow to be still the stable and default one. Now, renaming the old workflow to "old-xyz" and changing the behaviour of an established workflow sends a very different message, makes this a breaking change (which it would not be otherwise) and therefor goes against the idea of trying to improve the perceived quality of our software. In the backend, we embraced the policy of not implementing any breaking changes without deprecating the old behaviour and giving users a grace period of some weeks.
> 
> I think I would like this PR a lot better, if it left `excel2lists` and `excel2json` unchanged, and exposed either `excel2lists-experimental` or `excel2lists-new` or `excel2lists --experimental` or something like this. And just to be clear: The naming in code, I'm absolutely fine with. I'm really only referring to the naming used in the parser, which defines the CLI that is exposed to the users.
> 
> Apart from this one complaint: Great that this could be implemented so quickly! Thanks for your work!

Hey! Thanks for your comment, yesterday, this exactly occurred to me too and I was about to write a message about it.
I propose the following:
- Deprecation warning for regular `excel2json`
- Renaming into `new-excel2json`
- Opening up another project: `excel2json lists` refinement
    - look at removing complicated pandas logic
    - Possible simplification of the `Problem` classes
    - improving test coverage for the more higher level function
    - removing the old code and adapting the e2e tests
    - basically all the things I would have liked to do now before it was decided to do an emergency landing

What do you think?

File: src/dsp_tools/commands/excel2json/project.py
Comment (Nora-Olivia-Ammann) [reply src/dsp_tools/commands/excel2json/project.py:97]: This is the same way it appears on main, I didn't write it but I assume it serves a purpose.

Comment (BalduinLandolt) [reply src/dsp_tools/commands/excel2json/project.py:97]: sorry, missed the fact that this is one of the moved functions, not something you wrote.

**Comment by BalduinLandolt** [comment]
> > I only had a brief look. From what I can see in code, it looks ok.
> > However it is not exactly what we agreed upon in today's meeting: We said we'd accept the shortcut of not attempting to follow all quality assurance we should normally do, in order to get this finished out there, so you can prioritise other work without having to keep the branch open and the feature half-finished; and we agreed that this is acceptable, if we mark this as an experimental feature and consider the old workflow to be still the stable and default one. Now, renaming the old workflow to "old-xyz" and changing the behaviour of an established workflow sends a very different message, makes this a breaking change (which it would not be otherwise) and therefor goes against the idea of trying to improve the perceived quality of our software. In the backend, we embraced the policy of not implementing any breaking changes without deprecating the old behaviour and giving users a grace period of some weeks.
> > I think I would like this PR a lot better, if it left `excel2lists` and `excel2json` unchanged, and exposed either `excel2lists-experimental` or `excel2lists-new` or `excel2lists --experimental` or something like this. And just to be clear: The naming in code, I'm absolutely fine with. I'm really only referring to the naming used in the parser, which defines the CLI that is exposed to the users.
> > Apart from this one complaint: Great that this could be implemented so quickly! Thanks for your work!
> 
> Hey! Thanks for your comment, yesterday, this exactly occurred to me too and I was about to write a message about it. I propose the following:
> 
>     * Deprecation warning for regular `excel2json`
> 
>     * Renaming into `new-excel2json`
> 
>     * Opening up another project: `excel2json lists` refinement
>       
>       * look at removing complicated pandas logic
>       * Possible simplification of the `Problem` classes
>       * improving test coverage for the more higher level function
>       * removing the old code and adapting the e2e tests
>       * basically all the things I would have liked to do now before it was decided to do an emergency landing
> 
> 
> What do you think?

Good to hear that wasn't just me! :)

I would mainly focus on doing the renaming for now.  
If we add the deprecation warning already, then that still shifts the "default" to the new implementation, so I would hold off with that for the time being.  
And the follow-up project makes a lot of sense, I'll bring it up in today's PM meeting. Thanks!

File: src/dsp_tools/commands/excel2json/project.py
Comment (jnussbaum) [inline src/dsp_tools/commands/excel2json/project.py:210]: This line is not necessary:

```suggestion
```

File: test/unittests/commands/excel2json/test_new_lists.py
Comment (jnussbaum) [inline test/unittests/commands/excel2json/test_new_lists.py:37]: As mentioned, I would be okay if you lowercase the column titles for ease of use inside the python code. But I would prefer to have uppercase ID in the Excel

**Review by jnussbaum — APPROVED** [review]
LGTM. Only one of my comments is really important: I think that you made a mistake in this line of code, where you write the labels into the comments field:

```
return {"name": self.id_, "labels": self.labels, "comments": self.labels}
```

File: src/dsp_tools/commands/excel2json/models/list_node.py
Comment (Nora-Olivia-Ammann) [reply src/dsp_tools/commands/excel2json/models/list_node.py:108]: This is not an accident but I agree is pretty dumb. If we do not add a comment, then the validation of the json fails. I looked at how it is done now and this is the way. But I just noticed that the comments are not necessary for the nodes, so I will remove them there.

File: src/dsp_tools/commands/excel2json/project.py
Comment (Nora-Olivia-Ammann) [reply src/dsp_tools/commands/excel2json/project.py:210]: I did it because of a reference before assignement.

File: test/unittests/commands/excel2json/test_new_lists.py
Comment (Nora-Olivia-Ammann) [reply test/unittests/commands/excel2json/test_new_lists.py:37]: The excel remains unchanged, this is because when we clean the excel we make everything lower case so that this is not something that will raise a keyerror. Therefore the dfs that are used after the cleaning must also be lower case.

File: src/dsp_tools/commands/excel2json/new_lists.py
Comment (Nora-Olivia-Ammann) [reply src/dsp_tools/commands/excel2json/new_lists.py:90]: Thanks for spotting it, it is actually another error, I need the file name.

File: src/dsp_tools/commands/excel2json/new_lists.py
Comment (Nora-Olivia-Ammann) [reply src/dsp_tools/commands/excel2json/new_lists.py:177]: See other comment.

**Pull request merged**
